Expose complete Tracing APIs in the tracing toolkit (#442)
apache/skywalking-java: docs/en/setup/service-agent/java-agent/Application-toolkit-tracer.md

> # Span creating

Reviewer: Please consistent titles, current you have
- span creating
- Capture/continues
- Tag and log of spans

We should use consistent names. Such as, 
- Create Span
- Capture and Continuous Snapshot
- ...

Author: how about: 
* Create Span
* Inject/Extract Context Carrier
* Capture/Continue Context Snapshot
* Add Span's Tag and Log
* Async prepare/finish
* ActiveSpan
I try to make every title as verb-object phrase.

Reviewer: - Create Span
- Inject and Extract Context
- Capture and Continue Context
- Add Span's Tag and Log
- Set the span in async mode
- Access the Active Span

* Use `Tracer.createEntrySpan()` API to create entry span, and then use `SpanRef` to contain the reference of created span in agent kernel. The first parameter is operation name of span and the second parameter is the `ContextCarrierRef` instance which is the reference of contextcarrier in agent kernel. If the second parameter is not null, the process of creating entry span will do the extract operation which will be introduced in **inject/extract** scenario.

  ```java

>   import Tracer;

Reviewer: This import seems wrong? Why here without package name?

Author: I find that the old doc of tracing API only imports the class name, so I just write the Tracer class. So it should import the entire package, "org.apache.skywalking.apm.toolkit.trace.Tracer"?

Author: But actually, I find lots of package importing are lost in the Application-toolkit-tracer.md .

* Use `Tracer.stopSpan()` API to stop current span

  ```java
  import Tracer;
  ...
    
  Tracer.stopSpan();
  ```

# Inject/extract

The Inject/extract is to pass context information between different process. The `ContextCarrierRef` contains the reference of `ContextCarrier` and the `CarrierItemRef` contains the reference of CarrierItem. The `CarrierItem` instances compose a linked list. 

* Use `Tracer.inject()` to inject information of current context into carrier
* Use `Tracer.extract()` to extract info from contextCarrier. 
* Use `items()` of `ContextCarrierRef` instance to get head `CarrierItemRef` instance.  
* Use `hasNext()` of `CarrierItemRef` instance to judge if the `CarrierItemRef` has next item.
* Use `next()` of `CarrierItemRef` instance to get next item
* Use `getHeadKey` of `CarrierItemRef` instance to get key of current item
* Use `getHeadValue` of `CarrierItemRef` instance to get value of current item
* Use `setHeadValue` of `CarrierItemRef` instance to set value of current item

```java
/* 

> 	You can consider map as the message body, such as Http, MQ and RPC. 

Reviewer: ```suggestion
	You can consider map as the message header/metadata, such as Http, MQ and RPC. 
```

	This code block do the inject operation in one process and then pass the map message.
*/
ContextCarrierRef contextCarrierRef = new ContextCarrierRef();
Tracer.inject(contextCarrierRef);
Map<String, String> map = new HashMap<>();
CarrierItemRef next = contextCarrierRef.items();
while (next.hasNext()) {
    next = next.next();
    map.put(next.getHeadKey(), next.getHeadValue());
}
...
 
```

```java

> // This code block receive the map message and do the extract operation in another process. 

Reviewer: Why do we mention blocking?

Author: Because I want to remind users that these two code blocks are always used before sending the message and after receiving the message.

Author: So should I remove the "code block" pharse?

Reviewer: Please remove, block usually means it is a time cost operation.
And notice map should be header/metadata, not the message itself.

Author: Ok, I will remove the "code block", and the "message" has been changed to "header/metadata".

> // Receive the map in header/metadata and do the extract operation in another process. 

Reviewer: ```suggestion
// Receive the map representing a  header/metadata and do the extract operation in another process. 
```

> // Receive the map representing a  header/metadata and do the extract operation in another process. 

Reviewer: ```suggestion
// Receive the map representing a header/metadata and do the extract operation in another process. 
```
